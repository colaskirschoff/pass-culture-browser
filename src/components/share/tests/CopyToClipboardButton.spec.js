import React from 'react'
import { shallow } from 'enzyme'

import CopyToClipboardButton from '../CopyToClipboardButton'

const onClickMock = jest.fn()

describe('src | components | share | CopyToClipboardButton', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      // given
      const props = {
        onClick: onClickMock,
        value: 'Fake Value',
      }
      // when
      const wrapper = shallow(<CopyToClipboardButton {...props} />)

      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
  describe('functions', () => {
    describe('constructor', () => {
      it('should initialize state correctly', () => {
        // given
        const props = {
          onClick: onClickMock,
          value: 'Fake Value',
        }

        // when
        const wrapper = shallow(<CopyToClipboardButton {...props} />)
        const expected = {
          copied: false,
        }

        // then
        expect(wrapper.state()).toEqual(expected)
      })
    })
    describe('onCopy', () => {
      it('should change copied state to true', () => {
        // given
        const props = {
          onClick: onClickMock,
          value: 'Fake Value',
        }

        // when
        const wrapper = shallow(<CopyToClipboardButton {...props} />)
        wrapper.props().onCopy()
        const expected = {
          copied: true,
        }

        // then
        expect(wrapper.state()).toEqual(expected)
      })
    })
  })
})
