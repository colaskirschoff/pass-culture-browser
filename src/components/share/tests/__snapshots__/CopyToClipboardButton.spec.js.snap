// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | share | CopyToClipboardButton snapshot should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CopyToClipboardButton
    className=""
    onClick={[MockFunction]}
    value="Fake Value"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <button
        className="no-border no-background no-outline is-block "
        type="button"
      >
        <span>
          Copier le lien
        </span>
      </button>,
      "onCopy": [Function],
      "options": undefined,
      "text": "Fake Value",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            Copier le lien
          </span>,
          false,
        ],
        "className": "no-border no-background no-outline is-block ",
        "type": "button",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Copier le lien",
          },
          "ref": null,
          "rendered": "Copier le lien",
          "type": "span",
        },
        false,
      ],
      "type": "button",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <button
          className="no-border no-background no-outline is-block "
          type="button"
        >
          <span>
            Copier le lien
          </span>
        </button>,
        "onCopy": [Function],
        "options": undefined,
        "text": "Fake Value",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              Copier le lien
            </span>,
            false,
          ],
          "className": "no-border no-background no-outline is-block ",
          "type": "button",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Copier le lien",
            },
            "ref": null,
            "rendered": "Copier le lien",
            "type": "span",
          },
          false,
        ],
        "type": "button",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
